std/experimental/logger/filelogger.d

```diff
@@ -21,17 +28,37 @@ class FileLogger : Logger
       file can not be opened for writting an exception will be thrown.
       lv = The $(D LogLevel) for the $(D FileLogger). By default the
       $(D LogLevel) for $(D FileLogger) is $(D LogLevel.all).
+      createFileNameFolder = if yes and fn contains a folder name, this
+      folder will be created.
 
     Example:
     -------------
     auto l1 = new FileLogger("logFile");
     auto l2 = new FileLogger("logFile", LogLevel.fatal);
+    auto l3 = new FileLogger("logFile", LogLevel.fatal, CreateFolder.yes);
     -------------
     */
     this(in string fn, const LogLevel lv = LogLevel.all) @safe
     {
+        this(fn, lv, CreateFolder.yes);
+    }
+
+    this(in string fn, const LogLevel lv, CreateFolder createFileNameFolder) @safe
+    {
+        import std.file : exists, mkdirRecurse;
+        import std.path : dirName;
+
         super(lv);
         this.filename = fn;
+
+        if (createFileNameFolder == CreateFolder.yes)
+        {
+            auto d = dirName(this.filename);
+            mkdirRecurse(d);
+            assert(exists(d), "The folder the FileLogger should have been"
+                ~ " created in '" ~ d ~ "' could not be created.");
```

Comment:
You can use `std.conv.text` to make this a bit more efficient and legible (`text` uses an appender)